File: working hours register/Level 3/C#/worker_repository.cs
// WorkerRepository.cs
using System.Collections.Generic;
using System.Linq;

public class WorkerRepository
{
    private readonly Dictionary<string, Worker> _workers = new();

    public string AddWorker(string workerId, string position, int compensation)
    {
        if (_workers.ContainsKey(workerId))
            return "false";

        _workers[workerId] = new Worker(workerId, position, compensation);
        return "true";
    }

    public Worker? GetWorker(string workerId)
    {
        return _workers.TryGetValue(workerId, out var worker) ? worker : null;
    }

    public List<Worker> GetWorkersByPosition(string position)
    {
        return _workers.Values
            .Where(w => w.GetCurrentPosition() == position)
            .ToList();
    }
}

File: working hours register/Level 2/worker_repository.py
# worker_repository.py

class WorkerRepository:
    def __init__(self):
        self._workers = {}

    def add_worker(self, worker_id: str, position: str, compensation: int) -> str:
        if worker_id in self._workers:
            return "false"
        self._workers[worker_id] = Worker(worker_id, position, compensation)
        return "true"

    def get_worker(self, worker_id: str):
        return self._workers.get(worker_id)

    def get_workers_by_position(self, position: str) -> list:
        return [w for w in self._workers.values() if w.position == position]
